yt_dlp/downloader/hls.py

```diff
@@ -204,6 +205,11 @@ def is_ad_fragment_end(s):
                     })
                     media_sequence += 1
 
+                    # If the byte_range is truthy, reset it after appending a fragment that uses it
+                    if byte_range:
+                        byte_range_offset = byte_range.get('end', 0)
```

Comment:
If `byte_range` is truthy then we know it has an `end` key, or else if it doesn't then we should error because we messed up the code somewhere
```suggestion
                        byte_range_offset = byte_range['end']
```